File: HW/HW11/hw11.py
class hw11:
    def floodFill(self, image, sr, sc, color):
        currentColor = image[sr][sc]
        maxHeight = len(image)
        maxWidth = len(image[0])

        def dfs(sr, sc):
            if sr < 0 or sr >= maxHeight or sc < 0 or sc >= maxWidth or image[sr][sc] != currentColor or currentColor == color:
                return
            image[sr][sc] = color
            dfs(sr - 1, sc)
            dfs(sr, sc + 1)
            dfs(sr + 1, sc)
            dfs(sr, sc - 1)
        dfs(sr,sc)
        
        return image
        

File: HW/HW11/hw11.java
public class hw11{
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int currentColor = image[sr][sc];
        int maxHeight = image.length;
        int maxWidth = image[0].length;

        dfs(image, sr, sc, maxHeight, maxWidth, currentColor, color);
        return image;
    }

    private void dfs(int[][] image, int sr, int sc, int maxHeight, int maxWidth, int currentColor, int color){
        if (sr < 0 || sr >= maxHeight || sc < 0 || sc >= maxWidth || currentColor != image[sr][sc] || currentColor == color){
            return;
        }
        
        image[sr][sc] = color;
        dfs(image, sr - 1, sc, maxHeight, maxWidth, currentColor, color);
        dfs(image, sr, sc + 1, maxHeight, maxWidth, currentColor, color);
        dfs(image, sr + 1, sc, maxHeight, maxWidth, currentColor, color);
        dfs(image, sr, sc - 1, maxHeight, maxWidth, currentColor, color);
    }
}
